lib/location/method_gnss.c

```diff
@@ -59,6 +59,27 @@ BUILD_ASSERT(
 #if (defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS))
 #define AGPS_REQUEST_RECV_BUF_SIZE 3500
 #define AGPS_REQUEST_HTTPS_RESP_HEADER_SIZE 400
+/* Minimum time between two A-GPS data requests in seconds. */
+#define AGPS_REQUEST_MIN_INTERVAL (60 * 60)
+/* A-GPS data expiration threshold in seconds before requesting fresh data. An 80 minute threshold
+ * is used because it leaves enough time to try again after an hour if fetching of A-GPS data fails
+ * once. Also, because of how ephemeris validity times overlap, on average fresh ephemerides are
+ * needed every two hours with both 80 minute and shorter expiration thresholds.
+ */
+#define AGPS_EXPIRY_THRESHOLD (80 * 60)
+/* P-GPS data expiration threshold is zero seconds, because there's no overlap between
+ * predictions.
+ */
+#define PGPS_EXPIRY_THRESHOLD 0
+/* A-GPS minimum number of expired ephemerides to request all ephemerides. */
+#define AGPS_EPHE_MIN_COUNT 3
+/* P-GPS minimum number of expired ephemerides to trigger injection of predictions. With P-GPS
+ * predictions are not available for all satellites, especially when predictions are made further
+ * into the future, so it's natural that predictions are missing for some of the satellites.
+ */
+#define PGPS_EPHE_MIN_COUNT 12
```

Comment:
I selected this number so, that it would be higher than the number of ephes possibly missing in the predictions. The idea is, that new predictions do not need to be injected if GNSS already has the correct set (which usually does not have 32 ephemerides). Calling `nrf_cloud_pgps_notify_prediction()` would only cause the same set of ephemerides to be injected again.

The P-GPS library handles injection of the next set of predictions automatically with a timer. It also triggers download for new predictions when all previous predictions have been used. However, I realized now, that pre-emptive downloads do not probably work. Usually, `nrf_cloud_pgps_preemptive_updates()` has been called after injecting ephemerides, but the time based injection happens internally in the P-GPS library, so pre-emptive updates do not get triggered. I need to see where we should check if pre-emptive P-GPS updates are needed.

Reply:
Pre-emptive P-GPS updates are now checked when a GNSS fix is got, timeout is triggered or the location request is cancelled. I removed the check from prediction injection, because I don't think we need both.